File: ee-schkola/src/main/go/ee/schkola/finance/FinanceApiBase.go
package finance

import (
    "ee/schkola/person"
    "github.com/looplab/eventhorizon"
    "time"
)
type Expense struct {
    Purpose *ExpensePurpose `json:"purpose" eh:"optional"`
    Amount float64 `json:"amount" eh:"optional"`
    Profile *person.Profile `json:"profile" eh:"optional"`
    Date *time.Time `json:"date" eh:"optional"`
    Id eventhorizon.UUID `json:"id" eh:"optional"`
}

func NewExpense() (ret *Expense) {
    ret = &Expense{}
    return
}


type ExpensePurpose struct {
    Name string `json:"name" eh:"optional"`
    Description string `json:"description" eh:"optional"`
    Id eventhorizon.UUID `json:"id" eh:"optional"`
}

func NewExpensePurpose() (ret *ExpensePurpose) {
    ret = &ExpensePurpose{}
    return
}


type Fee struct {
    Student *person.Profile `json:"student" eh:"optional"`
    Amount float64 `json:"amount" eh:"optional"`
    Kind *FeeKind `json:"kind" eh:"optional"`
    Date *time.Time `json:"date" eh:"optional"`
    Id eventhorizon.UUID `json:"id" eh:"optional"`
}

func NewFee() (ret *Fee) {
    ret = &Fee{}
    return
}


type FeeKind struct {
    Name string `json:"name" eh:"optional"`
    Amount float64 `json:"amount" eh:"optional"`
    Description string `json:"description" eh:"optional"`
    Id eventhorizon.UUID `json:"id" eh:"optional"`
}

func NewFeeKind() (ret *FeeKind) {
    ret = &FeeKind{}
    return
}

File: ee-schkola/src/main/ts/ee/schkola/finance/FinanceApiBase.ts
import {Profile} from "../person/PersonApiBase"
import {SchkolaBase} from "../SharedApiBase"


export class Expense extends SchkolaBase {
    readonly purpose: ExpensePurpose
    readonly amount: number
    readonly profile: Profile
    readonly date: Date

}


export class ExpensePurpose extends SchkolaBase {
    readonly name: string
    readonly description: string

}


export class Fee extends SchkolaBase {
    readonly student: Profile
    readonly amount: number
    readonly kind: FeeKind
    readonly date: Date

}


export class FeeKind extends SchkolaBase {
    readonly name: string
    readonly amount: number
    readonly description: string

}









